card-validation/src/components/CardForm/index.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState } from 'react';
+import { Formik, Form, Field } from 'formik';
+import * as Yup from 'yup';
+import Card from '../Card';
+import './index.scss';
+
+const ValidationSchema = Yup.object().shape({
+  name: Yup.string()
+    .matches(/^((?:[A-Za-z]+ ?){1,3})$/, 'enter a valid name')
+    .required('required'),
+  cardnumber: Yup.string()
+    .matches(/^4[0-9]{12}(?:[0-9]{3})?$/, 'enter a valid card number')
+    .required('required'),
+  expireMM: Yup.string().required('required'),
+  expireYY: Yup.string().required('required'),
+  securitycode: Yup.string()
+    .min(3, 'too short')
+    .max(4, 'too long')
+    .required('required'),
+});
+
+export default function CardForm({ onSubmit }) {
+  const [side, setSide] = useState('front');
+
+  const handleSubmit = (values) => {
+    console.log('values', values);
+    onSubmit(values);
+  };
+
+  return (
+    <Formik
+      initialValues={{
```

Comment:
Due to readability it could be better to move it even out of the component to make the jsx part shorter and easier to read due to fitting the screen